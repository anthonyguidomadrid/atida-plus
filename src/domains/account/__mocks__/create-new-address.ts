export const createNewAddress = {
  redirect: 'some-path',
  salutation: 'string',
  firstName: 'string',
  lastName: 'string',
  address1: 'string',
  address3: 'string',
  zipCode: 'string',
  city: 'string',
  country: 'string',
  iso2Code: 'string',
  phone: 'string',
  isDefaultShipping: true,
  isDefaultBilling: true
}

export const createNewAddressPayload = {
  data: {
    type: 'customers',
    attributes: {
      salutation: 'string',
      firstName: 'string',
      lastName: 'string',
      address1: 'string',
      address3: 'string',
      zipCode: 'string',
      city: 'string',
      country: 'string',
      iso2Code: 'string',
      phone: 'string',
      isDefaultShipping: true,
      isDefaultBilling: true
    }
  }
} as const

export const UpdatedJWTAndRefreshTokens = {
  refreshToken: 'newRefreshToken',
  JWT: 'newJWT',
  expiresIn: '12345'
}

export const sprykerUpdatedJWTAndRefreshTokens = {
  data: {
    type: '',
    id: '',
    attributes: {
      tokenType: 'Bearer',
      expiresIn: '12345',
      accessToken: 'newJWT',
      refreshToken: 'newRefreshToken',
      idCompanyUser: ''
    },
    links: {
      self: ''
    }
  },
  links: {
    self: ''
  }
}
