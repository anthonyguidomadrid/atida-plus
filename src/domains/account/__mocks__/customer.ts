export const createCustomer = {
  salutation: 'Mr',
  firstName: 'Test',
  lastName: 'McTest',
  email: 'some@email.com',
  password: 'somePassword1!',
  confirmPassword: 'somePassword1!',
  acceptedTerms: true,
  emailNotification: true,
  phone: '654654654',
  company: null,
  surcharge: false,
  addresses: {
    addresses: [
      {
        salutation: 'Mr',
        firstName: 'Test',
        lastName: 'McTest',
        address1: 'Brugstraat 123',
        zipCode: '1011 AD',
        city: 'Amsterdam',
        iso2Code: 'PT',
        phone: '088273791298432',
        isDefaultShipping: true,
        isDefaultBilling: true
      }
    ]
  }
}

export const sprykerCustomer = {
  data: {
    type: 'customers',
    id: 'some-id',
    attributes: {
      salutation: 'Mr',
      firstName: 'Test',
      lastName: 'McTest',
      email: 'some@email.com',
      password: 'somePassword1!',
      confirmPassword: 'somePassword1!',
      acceptedTerms: true,
      emailNotification: true,
      company: null,
      surcharge: false
    }
  }
} as const

export const sprykerCustomerWithAddresses = {
  data: {
    ...sprykerCustomer,
    attributes: {
      ...sprykerCustomer.data.attributes
    },
    relationships: {
      addresses: {
        data: [
          { type: 'addresses', id: 'address-id-1' },
          { type: 'addresses', id: 'address-id-2' }
        ]
      }
    }
  },
  included: [
    {
      type: 'addresses',
      id: 'address-id-1',
      attributes: {
        salutation: 'some-salutation',
        firstName: 'some-name',
        lastName: 'some-last-name',
        address1: 'some-address',
        address2: 'some-address2',
        address3: 'some-address3',
        zipCode: 'some-zip-code',
        city: 'some-city',
        country: 'some-country',
        iso2Code: 'some-iso-code',
        company: 'some-company',
        phone: 'some-phone',
        isDefaultShipping: true,
        isDefaultBilling: true,
        isTaxExempt: false
      }
    },
    {
      type: 'addresses',
      id: 'address-id-2',
      attributes: {
        salutation: 'some-other-salutation',
        firstName: 'some-other-name',
        lastName: 'some-other-last-name',
        address1: 'some-other-address',
        address2: 'some-other-address2',
        address3: 'some-other-address3',
        zipCode: 'some-other-zip-code',
        city: 'some-other-city',
        country: 'some-other-country',
        iso2Code: 'some-other-iso-code',
        company: 'some-other-company',
        phone: 'some-other-phone',
        isDefaultShipping: false,
        isDefaultBilling: false,
        isTaxExempt: false
      }
    }
  ]
}
