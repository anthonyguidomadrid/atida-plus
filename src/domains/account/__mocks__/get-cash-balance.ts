export const balance = {
  total: 20
}
