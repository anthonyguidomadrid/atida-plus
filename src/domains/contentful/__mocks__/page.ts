export const contentfulPage = {
  fields: {
    title: 'some title',
    slug: '/some-slug',
    pageType: 'Category',
    referencedContent: {
      fields: {
        title: 'Medicines',
        level: 0,
        id: 'medicines',
        subcategories: [
          {
            fields: {
              title: 'Digestion',
              level: 1,
              id: 'medicines_digestion',
              color: {
                fields: {
                  ref: 'category-beauty'
                }
              },
              image: {
                fields: {
                  title: 'Beauty Category Image',
                  file: {
                    url:
                      '//images.ctfassets.net/7g2w796onies/2eBa8I8ATi8KEN5Sf4Ubu6/991b2273ce2b8b5159e8147a08751371/category-header-sample.png',
                    fileName: 'category-header-sample.png',
                    contentType: 'image/png'
                  }
                }
              }
            }
          },
          {
            fields: {
              title: 'Allergy',
              level: 1,
              id: 'medicines_allergy',
              color: {
                fields: {
                  ref: 'primary-caribbean-green-light'
                }
              }
            }
          },
          {
            fields: {
              title: 'Diabetes',
              level: 1,
              id: 'medicines_diabetes',
              color: {
                fields: {
                  ref: 'primary-caribbean-green-light'
                }
              }
            }
          }
        ],
        color: {
          fields: {
            ref: 'category-beauty'
          }
        },
        image: {
          fields: {
            title: 'Beauty Category Image',
            file: {
              url:
                '//images.ctfassets.net/7g2w796onies/2eBa8I8ATi8KEN5Sf4Ubu6/991b2273ce2b8b5159e8147a08751371/category-header-sample.png',
              details: [Object],
              fileName: 'category-header-sample.png',
              contentType: 'image/png'
            }
          }
        }
      }
    },
    contentBlocks: {},
    seo: {}
  }
}
