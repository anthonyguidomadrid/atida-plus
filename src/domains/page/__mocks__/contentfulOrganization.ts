const organization = {
  name: 'Atida',
  description: 'Online pharmacy',
  telephone: '123456789',
  email: 'hello@atida.com',
  sameAs: ['https://www.facebook.com', 'https://www.twitter.com'],
  imageUrl: 'logo.png',
  id: 'atida'
}

export const contentfulOrganization = {
  items: [{ fields: organization }]
}

export const organizationNormalized = organization
