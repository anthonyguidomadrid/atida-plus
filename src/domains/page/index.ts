export * from './helpers'
export * from './sagas'
export * from './selectors'
export * from './slices'
export * from './types'
