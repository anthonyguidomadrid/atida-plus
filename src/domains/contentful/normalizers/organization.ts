import type { EntryCollection } from 'contentful'
import { Organization, ContentfulOrganization } from '~domains/page'
import { removeUndefinedPropertiesFromObject } from '~helpers'

const normalizeOrganizationRest = (
  organizations?: EntryCollection<ContentfulOrganization>
): Organization | undefined => {
  const organization = organizations?.items?.[0]

  if (!organization) return undefined

  return removeUndefinedPropertiesFromObject({
    name: organization.fields.name,
    description: organization.fields.description,
    telephone: organization.fields.telephone,
    email: organization.fields.email,
    sameAs: organization.fields.sameAs,
    imageUrl: organization.fields.imageUrl,
    id: organization.fields.id,
    address: organization.fields.address?.fields
  })
}

export const normalizeOrganization = (
  organization?: EntryCollection<ContentfulOrganization>
): Organization | undefined => normalizeOrganizationRest(organization)
