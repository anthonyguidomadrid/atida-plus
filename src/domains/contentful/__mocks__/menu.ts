import { BLOCKS } from '@contentful/rich-text-types'
import type { DeepPartial } from '@reduxjs/toolkit'
import type { Asset } from 'contentful'
import {
  ContentfulCategory,
  ContentfulColor,
  ContentfulIcon,
  ContentfulLink
} from '../normalizers'
import { ContentfulMenu, Menu } from '../normalizers/menu'
import { ContentfulMenuItem } from '../normalizers/menu-item'

export const contentfulMenu: DeepPartial<ContentfulMenu> = {
  fields: {
    title: 'Some title',
    id: 'some id',
    items: [
      {
        fields: {
          id: 'Medicine',
          link: {
            fields: {
              title: 'Medicines',
              slug: '/medicines',
              pageType: 'Category',
              referencedContent: {
                fields: {
                  title: 'Medicines',
                  level: 0,
                  id: 'medicines',
                  color: {
                    fields: {
                      ref: 'category-beauty'
                    }
                  } as ContentfulColor,
                  image: {
                    fields: {
                      title: 'Beauty Category Image',
                      description: '',
                      file: {
                        url:
                          '//images.ctfassets.net/7g2w796onies/2eBa8I8ATi8KEN5Sf4Ubu6/991b2273ce2b8b5159e8147a08751371/category-header-sample.png',
                        fileName: 'category-header-sample.png',
                        contentType: 'image/png'
                      }
                    }
                  } as Asset
                }
              } as ContentfulCategory
            }
          },
          submenu: {
            fields: {
              id: '',
              title: 'Medicine Submenu',
              items: [
                {
                  fields: {
                    id: 'Diabetes',
                    link: {
                      fields: {
                        title: 'Diabetes',
                        slug: 'medicines/diabetes',
                        pageType: 'Category',
                        referencedContent: {
                          fields: {
                            title: 'Diabetes',
                            level: 1,
                            id: 'medicines_diabetes',
                            color: {
                              fields: {
                                ref: 'primary-caribbean-green-light'
                              }
                            }
                          }
                        }
                      }
                    }
                  }
                },
                {
                  fields: {
                    id: 'Alergies',
                    link: {
                      fields: {
                        title: 'Allergy & Hay fever',
                        slug: 'medicines/allergy',
                        pageType: 'Category',
                        referencedContent: {
                          fields: {
                            title: 'Allergy',
                            level: 1,
                            id: 'medicines_allergy',
                            color: {
                              fields: {
                                ref: 'primary-caribbean-green-light'
                              }
                            }
                          }
                        }
                      }
                    }
                  }
                },
                {
                  fields: {
                    id: 'Digestion',
                    link: {
                      fields: {
                        title: 'Digestion',
                        slug: 'medicines/digestion',
                        pageType: 'Category',
                        referencedContent: {
                          fields: {
                            title: 'Digestion',
                            level: 1,
                            id: 'medicines_digestion',
                            color: {
                              fields: {
                                ref: 'category-beauty'
                              }
                            },
                            image: {
                              fields: {
                                description: '',
                                title: 'Beauty Category Image',
                                file: {
                                  url:
                                    '//images.ctfassets.net/7g2w796onies/2eBa8I8ATi8KEN5Sf4Ubu6/991b2273ce2b8b5159e8147a08751371/category-header-sample.png',
                                  fileName: 'category-header-sample.png',
                                  contentType: 'image/png'
                                }
                              }
                            }
                          }
                        }
                      }
                    }
                  }
                }
              ]
            }
          }
        }
      } as ContentfulMenuItem,
      {
        fields: {
          id: 'Homeopathy and natural products',
          link: {
            fields: {
              label: '',
              url: '/a-different-url'
            }
          } as ContentfulLink
        }
      } as ContentfulMenuItem,
      {
        fields: {
          id: 'Terms and Conditions',
          link: ({
            fields: {
              label: 'Terms & Conditions',
              url: '/terms',
              content: {
                data: {},
                content: [
                  {
                    data: {},
                    content: [
                      {
                        data: {},
                        marks: [],
                        value: 'Including billing and pre-payment',
                        nodeType: 'text'
                      }
                    ],
                    nodeType: BLOCKS.PARAGRAPH
                  }
                ],
                nodeType: BLOCKS.DOCUMENT
              },
              icon: {
                fields: {
                  ref: 'PaymentSepa'
                }
              } as ContentfulIcon
            }
          } as unknown) as ContentfulLink
        }
      } as ContentfulMenuItem
    ]
  }
}

export const normalizedMenu: Menu = {
  title: 'Some title',
  id: 'some id',
  items: [
    {
      id: 'Medicine',
      link: {
        label: 'Medicines',
        url: '/medicines',
        category: {
          level: 0,
          id: 'medicines',
          color: 'category-beauty',
          image: {
            description: '',
            title: 'Beauty Category Image',
            url:
              '//images.ctfassets.net/7g2w796onies/2eBa8I8ATi8KEN5Sf4Ubu6/991b2273ce2b8b5159e8147a08751371/category-header-sample.png',
            type: 'image/png'
          }
        }
      },
      submenu: {
        id: '',
        title: 'Medicine Submenu',
        items: [
          {
            id: 'Diabetes',
            link: {
              label: 'Diabetes',
              url: '/medicines/diabetes',
              category: {
                level: 1,
                id: 'medicines_diabetes',
                color: 'primary-caribbean-green-light'
              }
            }
          },
          {
            id: 'Alergies',
            link: {
              label: 'Allergy & Hay fever',
              url: '/medicines/allergy',
              category: {
                level: 1,
                id: 'medicines_allergy',
                color: 'primary-caribbean-green-light'
              }
            }
          },
          {
            id: 'Digestion',
            link: {
              label: 'Digestion',
              url: '/medicines/digestion',
              category: {
                level: 1,
                id: 'medicines_digestion',
                color: 'category-beauty',
                image: {
                  description: '',
                  title: 'Beauty Category Image',
                  type: 'image/png',
                  url:
                    '//images.ctfassets.net/7g2w796onies/2eBa8I8ATi8KEN5Sf4Ubu6/991b2273ce2b8b5159e8147a08751371/category-header-sample.png'
                }
              }
            }
          }
        ]
      }
    },
    {
      id: 'Homeopathy and natural products',
      link: {
        content: '',
        label: '',
        url: '/a-different-url'
      }
    },
    {
      id: 'Terms and Conditions',
      link: {
        label: 'Terms & Conditions',
        url: '/terms',
        icon: 'PaymentSepa',
        content: '<p>Including billing and pre-payment</p>'
      }
    }
  ]
}

export const headerNavigationLeft: Menu = {
  title: 'Header Navigation Left',
  items: [
    {
      id: 'Products',
      link: {
        label: 'Products',
        url: '',
        icon: ''
      },
      submenu: normalizedMenu
    },
    {
      id: 'Promotions',
      link: {
        label: 'Promotions',
        url: '/promotions',
        icon: 'Percentage'
      }
    },
    {
      id: 'Brands',
      link: {
        label: 'Brands',
        url: '/brands',
        icon: 'Shop'
      }
    }
  ],
  id: 'header-navigation-left'
}

export const headerNavigationRight: Menu = {
  title: 'Header Navigation Right',
  items: [
    {
      id: 'Blog',
      link: {
        label: 'Blog',
        url: '/blog',
        icon: 'NavAdvice24'
      }
    },
    {
      id: 'FAQ',
      link: {
        label: 'FAQ',
        url: '/faq',
        icon: ''
      }
    },
    {
      id: 'Contact Us Header',
      link: {
        label: 'Contact',
        url: '/contact',
        icon: 'Telephone24'
      }
    }
  ],
  id: 'header-navigation-right'
}
