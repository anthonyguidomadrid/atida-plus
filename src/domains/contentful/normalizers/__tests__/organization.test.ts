import type { EntryCollection } from 'contentful'
import { ContentfulOrganization } from '~domains/page'
import { contentfulOrganization } from '~domains/page/__mocks__/contentfulOrganization'
import { normalizeOrganization } from '../organization'

describe(normalizeOrganization, () => {
  it('normalizes the data', () => {
    const normalizedData = normalizeOrganization(
      contentfulOrganization as EntryCollection<ContentfulOrganization>
    )
    expect(normalizedData).toMatchSnapshot()
  })

  it('returns empty if there is no data', () => {
    expect(normalizeOrganization(undefined)).toEqual(undefined)
  })

  it('returns empty if there are no items', () => {
    expect(
      normalizeOrganization(({
        items: undefined
      } as unknown) as EntryCollection<ContentfulOrganization>)
    ).toEqual(undefined)
  })

  it('returns empty if there are zero items', () => {
    expect(
      normalizeOrganization(({
        items: []
      } as unknown) as EntryCollection<ContentfulOrganization>)
    ).toEqual(undefined)
  })

  it('returns empty if there is no first item', () => {
    expect(
      normalizeOrganization(({
        items: [undefined]
      } as unknown) as EntryCollection<ContentfulOrganization>)
    ).toEqual(undefined)
  })

  it('returns partial data', () => {
    expect(
      normalizeOrganization(({
        items: [{ fields: { name: 'Atida' } }]
      } as unknown) as EntryCollection<ContentfulOrganization>)
    ).toEqual({
      name: 'Atida'
    })
  })
})
