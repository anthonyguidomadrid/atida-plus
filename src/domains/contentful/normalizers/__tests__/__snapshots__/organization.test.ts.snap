// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`normalizeOrganization normalizes the data 1`] = `
Object {
  "description": "Online pharmacy",
  "email": "hello@atida.com",
  "id": "atida",
  "imageUrl": "logo.png",
  "name": "Atida",
  "sameAs": Array [
    "https://www.facebook.com",
    "https://www.twitter.com",
  ],
  "telephone": "123456789",
}
`;
